packages/server/modules/workspaces/services/management.ts

```diff
@@ -166,13 +169,20 @@ export const createWorkspaceFactory =
       discoverabilityEnabled: false
     }
     await upsertWorkspace({ workspace })
+
     // assign the creator as workspace administrator
+    const role = Roles.Workspace.Admin
     await upsertWorkspaceRole({
       userId,
-      role: Roles.Workspace.Admin,
+      role,
       workspaceId: workspace.id,
       createdAt: new Date()
     })
+    await ensureValidWorkspaceRoleSeat({
```

Comment:
Refactored seat creation to make it easier to work with - instead of it being created in an event handler somewhere, it always gets created/fixed on every role insert/update. This ensures that RoleUpdated event listeners always have a valid seat to work with